ext/soap/soap.c

```diff
@@ -2368,6 +2368,8 @@ PHP_METHOD(SoapClient, SoapClient)
 				Z_TYPE_P(tmp) == IS_RESOURCE) {
 			context = php_stream_context_from_zval(tmp, 1);
 			Z_ADDREF_P(tmp);
+		} else {
+			context = php_stream_context_alloc(); 
```

Comment:
Now that the condition includes `!context ||`, is this part of the change still necessary?